cmake/Modules/FindSYCL/run_sycl.cmake

```diff
@@ -37,9 +37,14 @@ set(SYCL_host_compiler_flags "-fsycl-host-compiler-options=")
 set(SYCL_include_args)
 
 foreach(dir ${SYCL_include_dirs})
-  # Extra quotes are added around each flag to help SYCL parse out flags with spaces.
-  list(APPEND SYCL_include_args "-I${dir}")
-  string(APPEND SYCL_host_compiler_flags "-I${dir} ")
+  # Args with spaces need quotes around them to get them to be parsed as a single argument.
+  if(dir MATCHES " ")
```

Comment:
Is it a special handle for MSVC or Windows CMAKE? if so, please add some comments for readability.

Reply:
It's OS-agnostic to handle paths with spaces with double quotes. Issue was caught on Windows probably because default oneAPI path on Windows includes spaces.